streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -236,4 +239,81 @@ trait ZStreamPlatformSpecificConstructors { self: ZStream.type =>
    */
   final def fromJavaStreamTotal[A](stream: => ju.stream.Stream[A]): ZStream[Any, Nothing, A] =
     ZStream.fromJavaIteratorTotal(stream.iterator())
+
+  /**
+   * Create a stream of accepted connection from server socket
+   * Emit socket `Connection` from which you can read / write and ensure it is closed after it is used
+   */
+  def fromSocketServer(
+    port: Int,
+    host: Option[String] = None
+  ): ZStream[Blocking, Throwable, Managed[Throwable, Connection]] =
```

Comment:
We're not using `blocking` anywhere, but it's possible that we should: we need to audit the methods we're calling on the channel and socket and see which ones might block. `AsynchronousServerSocketChannel.open/bind` look suspicious, for example, so we might have to switch to the blocking pool there.

The methods that take a completion handler look safe though.

Reply:
i forgot to cleanup the method. Good catch `open/bind` should be wrap in blocking pool.